packages/typespec-ts/test/modularIntegration/authApiKey.spec.ts

```diff
@@ -29,7 +29,7 @@ describe("ApiKeyClient Classical Client", () => {
       await client.invalid();
       assert.fail("Expected an exception to be thrown.");
     } catch (err: any) {
-      assert.strictEqual(err.error, "invalid-api-key");
+      assert.strictEqual(err.message, "Unexpected status code 403");
```

Comment:
for this case, the return error is { body: { error: "invalid-api-key" }}, but we don't think the error should be just be a string in the following code.  I wonder if we should work around this case or leave it ? @deyaaeldeen 
```ts
    const internalError = (result.body as any).error || result.body || result;
    const message = `Unexpected status code ${result.status}`;
    throw new RestError(internalError.message ?? message, {
      statusCode: Number(result.status),
      code: internalError.code,
      request: result.request,
    });
```

Reply:
What service is this? I think we should have a discussion with them about how their error responses are not conforming to the Azure REST API design guidelines.

I don't think other languages are doing anything special for this case based on the email thread I sent to you so I vote to leave it as is. /cc @xirzec @lmazuel

Reply:
This is a cadl ranch test case, I agree to leave it as is.